File: samplejavaapp/src/main/java/com/yoox/samplejavaapp/detail/DetailViewModel.java
package com.yoox.samplejavaapp.detail;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;
import com.yoox.net.Items;
import com.yoox.net.ItemsBuilder;
import com.yoox.net.Request;
import com.yoox.net.models.outbound.Item;
import com.yoox.sylvia.androidcallback.AndroidExecutor;
import com.yoox.sylvia.androidcallback.Callback;

public class DetailViewModel extends ViewModel {

    private final Items items = new ItemsBuilder("IT").build();

    private final MutableLiveData<Item> itemLiveData = new MutableLiveData<>();
    private final MutableLiveData<String> errorLiveData = new MutableLiveData<>();

    public LiveData<Item> getItem() {
        return itemLiveData;
    }

    public LiveData<String> getError() {
        return errorLiveData;
    }


    void loadItem(String id) {

        Request<Item> request = items.get(id);


        AndroidExecutor.execute(request, new Callback<Item>() {
            @Override
            public void onComplete(Item result) {
                DetailViewModel.this.itemLiveData.setValue(result);
            }

            @Override
            public void onError(@NonNull Throwable t) {
                DetailViewModel.this.errorLiveData.setValue(t.toString());
            }
        });

    }

}

File: samplekotlinapp/src/main/java/com/yoox/samplekotlinapp/detail/DetailViewModel.kt
package com.yoox.samplekotlinapp.detail

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.yoox.net.ItemsBuilder
import com.yoox.net.models.outbound.Item
import com.yoox.samplekotlinapp.common.ScopedViewModel
import kotlinx.coroutines.launch

class DetailViewModel : ScopedViewModel() {
    private val items = ItemsBuilder("IT").build()

    private val itemLiveData = MutableLiveData<Item>()
    private val errorLiveData = MutableLiveData<String>()

    val item: LiveData<Item>
        get() = itemLiveData

    val error: LiveData<String>
        get() = errorLiveData

    internal fun loadItem(id: String) {
        scope.launch {
            try {
                val request = items.get(id)
                val result = request.execute()
                itemLiveData.value = result
            } catch (t: Throwable) {
                errorLiveData.value = t.toString()
            }
        }
    }
}
